src/ThreeWayMerge.php

```diff
@@ -6,52 +6,316 @@
+                throw new Exception("A whole new conflict arised");
+            } elseif ($mincount == $count_remote
+                && $ancestor[$key] != $local[$key]) {
+                throw new Exception("A whole new conflict arised");
+            }
+        }
+            return $merged;
+    }
+
+    /**
+     * Method to deal with the case where Addition and removal.
+     * takes place simuntaneously.
+     * Example: 2 lines in ancestor, 3 lines in local.
+     * and 1 line in remote.
+     *
+     * @param array $ancestor
+     * @param array $local
+     * @param array $remote
+     * @param int   $count_ancestor
+     * @param int   $count_local
+     * @param int   $count_remote
+     *
+     * @return array
+     * @throws Exception
+     */
+    protected function linesAddedRemovedAndModified(
+        array $ancestor,
+        array $local,
+        array $remote,
+        $count_ancestor,
+        $count_local,
+        $count_remote
+    ) {
+        $merged = [];
+        $count_array = [$count_ancestor, $count_local, $count_remote];
+        sort($count_array);
+        $mincount = min($count_local, $count_ancestor, $count_remote);
+        $maxcount = max($count_local, $count_ancestor, $count_remote);
+        
+        // First for loop compares all 3 nodes and returns updated node.
+        for ($key = 0; $key < $mincount; $key++) {
+            if ($ancestor[$key] == $local[$key]) {
+                $merged[$key] = $remote[$key];
+            } elseif ($ancestor[$key] == $remote[$key]
+                || $local[$key] == $remote[$key]) {
+                $merged[$key] = $local[$key];
+            } else {
+                throw new Exception("A conflict has occured");
+            }
+        }
+        
+        // Second for loop compares 2 nodes and if they are identical
+        // add the changes to new array otherwise throw conflict exception.
+        for ($key = $mincount; $key < $count_array[1]; $key++) {
+            if ($count_ancestor == $mincount
+                && ($count_remote == $maxcount || $count_local == $maxcount)
+            ) {
+                if ($local[$key] == $remote[$key]) {
                     $merged[$key] = $local[$key];
+                } else {
+                    throw new Exception("A conflict has occured");
+                }
+            } elseif ($count_local == $mincount
+                && ($count_ancestor == $maxcount
+                    || $count_remote == $maxcount)
+            ) {
+                if ($ancestor[$key] == $remote[$key]) {
+                    if (!isset($local[$key])) {
+                        unset($merged[$key]);
+                    }
+                } else {
+                    throw new Exception("A conflict has occured");
+                }
+            } elseif ($count_remote == $mincount
+                && ($count_ancestor == $maxcount
+                    || $count_local == $maxcount)) {
+                if ($local[$key] == $ancestor[$key]) {
```

Comment:
For all conditions above that span multiple lines, let's do like this instead:
```php
} elseif (
  $count_remote == $mincount
  && ($count_ancestor == $maxcount || $count_local == $maxcount)
) {
```
Notice above how I kept the condition within the `( )` on a single line. That'll be a lot more clear, since it still fits within the line limit.